File: lesson-5/5B.py
n, k = map(int, input().split())
numbers = list(map(int, input().split()))

prefix_sum = [0] * (n + 1)

for i in range(1, n + 1):
    prefix_sum[i] = prefix_sum[i - 1] + numbers[i - 1]

counter = 0
length = len(prefix_sum)

for i in range(length - 1):
    current = prefix_sum[i]
    pointer = i + 1

    while pointer < length:
        diff = prefix_sum[pointer] - current

        if diff > k:
            break
        
        if diff == k:
            counter += 1
        
        pointer += 1

print(counter)

File: lesson-5/5B.js
const fs = require('fs');

const solution = ([meta, cars]) => {
    const [n, k] = meta.split(' ').map(Number);
    const numbers = cars.split(' ').map(Number);
    const prefixSum = [0];

    for (let i = 1; i < n + 1; i += 1) {
        prefixSum[i] = prefixSum[i - 1] + numbers[i - 1];
    }

    let counter = 0;
    const len = prefixSum.length;

    for (let i = 0; i < len - 1; i += 1) {
        const current = prefixSum[i];
        let pointer = i + 1;

        while (pointer < len) {
            const diff = prefixSum[pointer] - current;
            console.log(diff);

            if (diff > k) {
                break;
            }

            if (diff === k) {
                counter += 1;
            }

            pointer += 1;
        }
    }

    console.log(counter);
};

const input = fs.readFileSync('input.txt', 'utf-8');
const data = input.split(/\r?\n/);

solution(data);